test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```diff
@@ -0,0 +1,400 @@
+            rLV27qwHeQIDAQABo4GOMIGLMA4GA1UdDwEB/wQEAwIEMDAUBgNVHSUEDTALBgkr
+            BgEEAYI3UAEwRAYJKoZIhvcNAQkPBDcwNTAOBggqhkiG9w0DAgICAIAwDgYIKoZI
+            hvcNAwQCAgCAMAcGBSsOAwIHMAoGCCqGSIb3DQMHMB0GA1UdDgQWBBRIyIzwInLJ
+            3B+FajVUFMACf1hrxjANBgkqhkiG9w0BAQUFAAOCAQEAfFt4rmUmWfCbbwi2mCrZ
+            Osq0lfVNUiZ+iLlEKga4VAI3sJZRtErnVM70eXUt7XpRaOdIfxjuXFpsgc37KyLi
+            ByCORLuRC0itZVs3aba48opfMDXivxBy0ngqCPPLQsyaN9K7WnpvYV1QxiudYwwU
+            8U5rFmzlwNLvc3XiyoGWaVZluk2DIJawQ5QYAU9/NMBBCbPHjTG7k0l4cpcEC+Ex
+            od3RlO6/MOYuK2WB4VTxKsV80EdA3ljlu7Td8P4movnrbB4rG4wpCpk05eREkg/5
+            Y54Ilo9m5OSAWtdx4yfS779eebLgUs3P+dk6EKwovXMokVveZA8cenIp3QkqSpeT
+            cQ==
+            -----END CERTIFICATE-----
+            "
+
+            $ers = $null
+            $recipient = [System.Management.Automation.CmsMessageRecipient] $certContent
+            $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+            $recipient.Certificates[0].Subject | Should Match 'CN=MyDataEnciphermentCert'
+    }
+
+    It "Verify wildcarded recipient resolution by path [Decryption]" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] ($certLocation + "*")
+        $recipient.Resolve($ExecutionContext.SessionState, "Decryption", [ref] $ers)
+
+        # Should have resolved single cert
+        $recipient.Certificates.Count | Should Be 1
+    }
+
+    It "Verify wildcarded recipient resolution by path [Encryption]" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] ($certLocation + "*")
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $recipient.Certificates.Count | Should Be 1
+    }
+
+    It "Verify resolution by directory" {
+        $protectedEventLoggingCertPath = Join-Path $TestDrive ProtectedEventLoggingDir
+        $null = New-Item -ItemType Directory $protectedEventLoggingCertPath -Force
+        Copy-Item $certLocation $protectedEventLoggingCertPath
+        Copy-Item $certLocation (Join-Path $protectedEventLoggingCertPath "SecondCert.pfx")
+        Copy-Item $certLocation (Join-Path $protectedEventLoggingCertPath "ThirdCert.pfx")
+
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $protectedEventLoggingCertPath
+        $recipient.Resolve($executionContext.SessionState, "Decryption", [ref] $ers)
+
+        $recipient.Certificates.Count | Should Be 1
+    }
+
+    It "Verify resolution by thumbprint" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $importedCert.Thumbprint
+        $recipient.Resolve($ExecutionContext.SessionState, "Decryption", [ref] $ers)
+
+        # "Should have certs from thumbprint in 'My' store"
+        $recipient.Certificates[0].Thumbprint | Should Be $importedCert.Thumbprint
```

Comment:
Good catch. I will add `$recipient.Certificates.Count | Should Be 1` before this test.